storage/redis_adapter.go

```diff
@@ -92,18 +94,24 @@ func NewRedisStorageAdapter(bufferSize int) (*redisAdapter, error) {
 }
 
 func (a *redisAdapter) start() {
-	mp := newMessagePipeliner(a.bufferSize, a.redisClient)
+	errCh := make(chan error)
+	mp := newMessagePipeliner(a.bufferSize, a.redisClient, errCh)
+	go func() {
+		for {
+			log.Println(<-errCh)
+		}
+	}()
```

Comment:
this goroutine will never exit. use a `select` statement to either:

- time out reception on `errCh` and `return`
- watch for closure of `errCh` (implies that the sender of `errCh` will need to close it)